frontend/lib/src/components/widgets/CustomComponent/ComponentInstance.tsx

```diff
@@ -374,10 +374,13 @@ export class ComponentInstance extends React.PureComponent<Props, State> {
         src = this.props.registry.getComponentURL(componentName, "index.html")
       }
 
+      // Create a URL object from window.location
+      const currentUrl = new URL(window.location.href)
+
       // Add streamlitUrl query parameter to src.
       src = queryString.stringifyUrl({
         url: src,
-        query: { streamlitUrl: window.location.href },
+        query: { streamlitUrl: currentUrl.origin + currentUrl.pathname },
       })
```

Comment:
Hi, so it seems like this is taking the currentUrl's origin and the pathname within declaring the src variable. However, it seems like shouldn't we just be not setting src to this `queryString.stringifyUrl({...})`?

Reply:
That's what we have done internally as a workaround, since there's nothing in the Streamlit sources that actually references this URL parameter.

_However_, there are a number of open-source custom components that appear to depend on this functionality, and I was hesitant to submit a PR that would break them. For example, https://github.com/andfanilo/streamlit-drawable-canvas/blob/e9c74ad9f58e8ae6d51c3b4320f1851f7858584c/streamlit_drawable_canvas/frontend/src/DrawableCanvas.tsx#L18 uses it to load the background image, and https://github.com/search?type=code&q=%2F%28%3F-i%29streamlitUrl%2F+language%3ATSX&p=1&l=TSX finds several other projects that use it. Presumably there's other closed-source code that depends on it as well.

It's worth noting that most of these use cases can probably also be resolved by passing an argument to the custom component, but that would rely on changing the logic in their code.